onnxruntime/core/providers/cuda/cuda_execution_provider.cc

```diff
@@ -59,8 +59,13 @@ ONNX_OPERATOR_KERNEL_EX(
 
 CUDAExecutionProvider::PerThreadContext::PerThreadContext(OrtDevice::DeviceId device_id, size_t cuda_mem_limit, ArenaExtendStrategy arena_extend_strategy) {
   CUDA_CALL_THROW(cudaSetDevice(device_id));
+  CUDA_CALL_THROW(cudaStreamCreateWithFlags(&stream_, cudaStreamNonBlocking));
+
   CUBLAS_CALL_THROW(cublasCreate(&cublas_handle_));
+  CUBLAS_CALL_THROW(cublasSetStream(cublas_handle_, stream_));
```

Comment:
Should we add an option to use legacy default stream for cublas/cudnn, since per-thread workspace may increase memory footprint? We can default it to off, but it would be nice to have in case someone runs into memory pressure.

Reply:
Seems conv kernels are using per-thread scratch buffer so no need to add option. We might consider that in future if memory footprint is a concern in multi-threaded inference.